libs/build_dod_dataset.py

```diff
@@ -105,6 +112,120 @@ def get_projections_df():
     "Unassigned": NULL_VALUE
 }
 
+def get_state_and_counties():
+    # go to the JHUDataset to get this, for w/e reason
+    timeseries = JHUDataset.local().timeseries()
+    timeseries = timeseries.get_subset(
+        AggregationLevel.COUNTY,
+        after=datetime.datetime(2020, 3, 7),
+        country="USA", state=None
+    )
+
+    counties_by_state = defaultdict(list) # why tho
+
+    county_keys = timeseries.county_keys()
+    for country, state, county, fips in county_keys:
+        counties_by_state[state].append((county, fips))
+
+    return counties_by_state
+
+def find_peak(projection, col):
+    peak_row = []
+    for row in projection:
+        if not peak_row or (float(row[col]) > float(peak_row[col])):
+            peak_row = row
+    return peak_row
+
+def get_county_projections():
+    # for each state in our data look at the results we generated via run.py
+    # to create the projections
+    intervention_type = NO_INTERVENTION # None, as requested
+
+    # get 16 and 32 days out from now
+    today = datetime.datetime.now()
+    sixteen_days = today + datetime.timedelta(days=16)
+    thirty_two_days = today + datetime.timedelta(days=32)
+
+    #save results in a list of lists, converted to df later
+    results = []
+
+    # get the state and fips so we can get the files (TODO: this is trash)
+    missing = 0
+    for state, counties in get_state_and_counties().items():
+        for county, fips in counties:
+            file_name = f"{state}.{fips}.{intervention_type}.json"
+            path = os.path.join(OUTPUT_DIR_COUNTIES, file_name)
+            # if the file exists in that directory then process
+            if os.path.exists(path):
+                with open(path, "r") as projections:
+                    # note that the projections have an extra column vs the web data
+                    projection =  simplejson.load(projections)
+                    hosp_16_days, short_fall_16_days = get_hospitals_and_shortfalls(projection, sixteen_days)
+                    hosp_32_days, short_fall_32_days = get_hospitals_and_shortfalls(projection, thirty_two_days)
+                    hospitalizations = [int(row[9]) for row in projection]
+                    deaths = [int(row[11]) for row in projection]
+                    new_deaths = list(range(len(deaths)))
+                    new_deaths[0] = 0
+                    for i in range(1, len(deaths)):
+                        new_deaths[i] = int(deaths[i]) - int(deaths[i-1])
+                    mean_hospitalizations = math.floor(statistics.mean(hospitalizations))
+                    mean_deaths = math.floor(statistics.mean(new_deaths))
+                    peak_hospitalizations_date = find_peak(projection, 9)[1]
```

Comment:
my main recommendation here would be to reference columns by name as opposed to by number

Reply:
cool these are json but made a constant for clarity